assets/js/apps/search/components/filters/date-filter/DateRangeFilter.jsx

```diff
@@ -0,0 +1,194 @@
+import React, { useState, useEffect } from 'react';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import PropTypes from 'prop-types';
+import Datetime from 'react-datetime';
+import moment from 'moment';
+
+// React Datetime requires CSS to work.
+import 'react-datetime/css/react-datetime.css';
+
+const isNone = (value) => {
+    // We need empty string to represent an empty field for text fields
+    return value === undefined || value === null || value === "";
+}
+
+const isValueEmpty = (value) => {
+    if (isNone(value)) {
+        return true;
+    }
+    const { start, end } = value;
+    // A number range filter value is empty if both values are null values.
+    return isNone(start) && isNone(end);
+};
+
+const toSubmitValue = localValue => {
+    // Replace empty string value with null to represent null parameter value.
+    if (!localValue) {
+        return null;
+    }
+    const submitValue = [];
+    if (!isNone(localValue.start)) {
+        if (typeof localValue.start == "object") {
+            submitValue.push({
+                op: ">=",
+                content: localValue.start.toISOString()
+            });
+        }
+    }
+    if (!isNone(localValue.end)) {
+        if (typeof localValue.end == "object") {
+            submitValue.push({
+                op: "<=",
+                content:localValue.end.toISOString()
```

Comment:
Codacy has a fix for the issue: [Missing space before value for key 'content'.](https://app.codacy.com/gh/mytardis/mytardis/pullRequest?prid=7933948)
```suggestion
                content: localValue.end.toISOString()
```